### PR TITLE
FAQ Update

### PR DESCRIPTION
**Pull request by kartik-gupta-ij**
[PREVIEW](https://deploy-preview-879--condescending-goldwasser-91acf0.netlify.app/documentation/faq/qdrant-fundamentals/)

The changes are pretty conservative. I added a few more basic questions and cleaned up the sections.

I believe the FAQ may not be the best idea in the long term. We need to fix the SSOT and then we won't need the FAQ.

### CODE REVIEW

File: qdrant-landing/content/documentation/faq/database-optimization.md [hunk 1]
```diff
@@ -34,11 +34,38 @@ As a result, the Qdrant process might use more memory than the minimum required
...
+### How can Qdrant I enhance search performance when applying filters to their queries?
+
+To improve search performance with filters in Qdrant:
+
+- **Utilize Payload Index:** Implement payload indexes for fields used in filtering conditions. This enhances point retrieval speed based on filter criteria.
+- **Consider Full-text Index:** For string payloads, utilize full-text indexes for word or phrase presence filtering, adjusting tokenization parameters as needed.
+- **Parameterized Integer Index:** Opt for parameterized integer indexes for specific range filter requirements, balancing between lookup and range operations.
+- **Leverage Vector Index:** Employ HNSW for dense vector indexing, adjusting parameters like `m` and `ef_construct` for improved search accuracy and efficiency.
+- **Sparse Vector Index:** For sparse vectors, ensure efficient indexing and search mechanisms. Configure memory or disk storage based on segment characteristics and query patterns.
+- **Filtrable Index Extension:** Enhance HNSW graph with additional edges based on payload values to optimize search efficiency with varying filter stringency.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/database-optimization.md:68]: is this dome kind of chatGPT generated?

Comment (kartik-gupta-ij) [reply qdrant-landing/content/documentation/faq/database-optimization.md:68]: I have actually taken the content from [here](https://qdrant.tech/documentation/concepts/filtering) and used ChatGPT to rephrase it, thereby mistakenly incorporating the unwanted additional content in the process.

I was planing to cross verify everything after completing the questions. I am sorry for the mess, fixing it right away.

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md [hunk 2]
```diff
@@ -81,3 +81,35 @@ We only guarantee compatibility if you update between consecutive versions. You
...
+Yes, a listener node can indeed receive data from all shards. However, it's important to note that a listener node doesn't participate in searches.
+
+###  When creating snapshots from the listener node, will they include data from all shards by default?
+
+No, snapshots from the listener node won't automatically include data from all shards. You'll need to manually replicate all shards to the listener node. You can use the `replicate_shard` API for this purpose.
+
+
+### If I want to use a listener node solely for data storage and scaling out my cluster, can I do so without affecting other nodes with query traffic?
+
+Absolutely. You can configure a listener node to act as a pure data node, thereby scaling out your cluster without impacting other nodes with query traffic.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:111]: we are going to deprecate Listener node and it is definitely not a frequently asked question

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md [hunk 3]
```diff
@@ -81,3 +81,35 @@ We only guarantee compatibility if you update between consecutive versions. You
 
 In case your version is older, we only guarantee compatibility between two consecutive minor versions. This also applies to client versions. Ensure your client version is never more than one minor version away from your cluster version.
 While we will assist with break/fix troubleshooting of issues and errors specific to our products, Qdrant is not accountable for reviewing, writing (or rewriting), or debugging custom code.
+
+### Which embedding models does FastEmbed support, and how does it utilize memory?
+
+You can find the list of supported models [here](https://qdrant.github.io/fastembed/examples/Supported_Models/).
+
+Memory usage in fastEmbed depends on several factors:
+1. **Size of the Text**: Everything, including the text and its vectors, is loaded into memory. As vectors are computed, RAM consumption increases accordingly.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:90]: this is not true, fastembed is able to stream encoding, it doesn't need all vectors and texts in ram

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md [hunk 4]
```diff
@@ -81,3 +81,35 @@ We only guarantee compatibility if you update between consecutive versions. You
...
+2. **Data Parallelism**: fastEmbed utilizes Python's multiprocessing to split large lists of strings into smaller ones and process them in parallel. While this enhances processing speed, it can also lead to higher RAM consumption.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:91]: where is the source of this info?

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md [hunk 5]
```diff
@@ -81,3 +81,35 @@ We only guarantee compatibility if you update between consecutive versions. You
...
+3. **Model Used**: Memory usage varies depending on the model used to embed your data.
+
+For optimal performance and memory management, consider these factors when using fastEmbed.
+
+### To ensure all shards are replicated to the listener node, can I use the "Creating new shard replicas" API right after bootstrapping a new cluster and loading data, but before clients start sending new requests?
+
+Yes, that's correct. Before clients start sending requests, you can use the "Creating new shard replicas" API to replicate all shards to the listener node. This ensures that all new data will be replicated to the data node, facilitating whole cluster backup or scaling out.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:98]: this is not frequently asked and we are going to deprecate listener nodes.

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md [hunk 6]
```diff
@@ -81,3 +81,35 @@ We only guarantee compatibility if you update between consecutive versions. You
...
+No, snapshots from the listener node won't automatically include data from all shards. You'll need to manually replicate all shards to the listener node. You can use the `replicate_shard` API for this purpose.
+
+
+### If I want to use a listener node solely for data storage and scaling out my cluster, can I do so without affecting other nodes with query traffic?
+
+Absolutely. You can configure a listener node to act as a pure data node, thereby scaling out your cluster without impacting other nodes with query traffic.
+
+### How does your cloud handle shard rebalancing when increasing the number of nodes?
+
+Our [cloud platform](https://cloud.qdrant.io) handles shard rebalancing when scaling out. However, it's essential to create enough shards beforehand to facilitate the scaling process effectively. For instance, if you have 3 nodes, it's advisable to choose 6 or 9 shards to allow for rebalancing upon extending your cluster, which wouldn't be possible with just 3 shards.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:115]: > handles shard rebalancing when scaling out

does it?

File: qdrant-landing/content/documentation/faq/database-optimization.md [hunk 7]
```diff
@@ -34,11 +34,30 @@ As a result, the Qdrant process might use more memory than the minimum required
...
 - **Usage of on-disk vector storage with slow disks** -- If you're using on-disk vector storage, ensure you have fast enough disks. We recommend using local SSDs with at least 50k IOPS. Read more about the influence of the disk speed on the search latency in the article about [Memory Consumption](../../../articles/memory-consumption/).
 - **Large limit or non-optimal query parameters** -- A large limit or offset might lead to significant performance degradation. Please pay close attention to the query/collection parameters that significantly diverge from the defaults. They might be the reason for the performance issues.
+
+### How can I optimize optimizer's configuration settings for better accuracy and speed, especially for large-scale collections?
+
+To optimize optimizer's config in Qdrant for better accuracy and speed, consider the following:
+
+- For low memory footprint with high-speed search, utilize vector quantization with disk storage for vectors and in-memory quantized vectors. Configure memmap_threshold and always_ram accordingly.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/database-optimization.md:49]: `memmap_threshold` is a param we are going to deprecate. Use `on_disk` instead

File: qdrant-landing/content/documentation/faq/database-optimization.md [hunk 8]
```diff
@@ -34,11 +34,30 @@ As a result, the Qdrant process might use more memory than the minimum required
...
+- To prioritize high precision with a low memory footprint, enable on-disk vectors and HNSW index. Adjust HNSW parameters for precision while considering disk IOPS.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/database-optimization.md:50]: does this answer the question? It feels like a copy-paste from https://qdrant.tech/documentation/guides/optimize/, why don't we just give a link to it?

File: qdrant-landing/content/documentation/faq/database-optimization.md [hunk 9]
```diff
@@ -34,11 +34,30 @@ As a result, the Qdrant process might use more memory than the minimum required
...
+- For high precision with high-speed search, focus on keeping data in RAM. Utilize quantization with re-scoring and adjust search-time parameters like hnsw_ef for accuracy and speed balance.
+- Balance latency vs. throughput based on your needs. Configure the number of segments to utilize CPU cores effectively for either minimizing latency or maximizing throughput.
+- Explore different quantization methods like scalar, binary, and product quantization based on accuracy, speed, and compression requirements.
+- Fine-tune quantization parameters such as quantile for scalar quantization to optimize search precision and memory usage.
+- Adjust storage modes to balance between memory footprint and search speed, considering factors like disk reads and storage type (RAM, SSD, or HDD).
+
+Read more about [optimizing](../../guides/optimize/), and [quantization](../../guides/quantization/) in Qdrant.
+
+### How can I enhance search performance when applying filters to their queries?
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/database-optimization.md:59]: their?

File: qdrant-landing/content/documentation/faq/database-optimization.md [hunk 10]
```diff
@@ -34,11 +34,30 @@ As a result, the Qdrant process might use more memory than the minimum required
...
+
+To enhance the search performance with the filters in qdrant, it's important to optimize the indexing strategies. Users can get a combination of vector and traditional indexes from qdrant, where the vector indexes reduce the time taken for the vector search and the payload indexes quicken the pace of filtering. Users need to strategically mark the fields as indexable as well as prioritize the fields that appear frequently in the filtering conditions in order to efficiently utilize the memory resources. Through thorough cosideration of the memory constraints as well as careful index configuration, users can effectively enhance search performance with filters in qdrant.
```
Comment (generall) [inline qdrant-landing/content/documentation/faq/database-optimization.md:61]: > as well as prioritize the fields that appear frequently in the filtering conditions

What does it mean to prioritize fields?

**Review by generall — CHANGES_REQUESTED** [review]

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md (hunk 4 above)
Comment (kartik-gupta-ij) [reply qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:91]: https://github.com/qdrant/fastembed/issues/222#issuecomment-2104016782

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md (hunk 6 above)
Comment (kartik-gupta-ij) [reply qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:115]: https://discord.com/channels/907569970500743200/1180029091320172616/1180082682034667531

File: qdrant-landing/content/documentation/faq/database-optimization.md (hunk 10 above)
Comment (kartik-gupta-ij) [reply qdrant-landing/content/documentation/faq/database-optimization.md:61]: Fields that are commonly used in filtering conditions and index them first as indexing cost resources

Comment (generall) [reply qdrant-landing/content/documentation/faq/database-optimization.md:61]: What the difference if you index it first or last?

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md (hunk 6 above)
Comment (generall) [reply qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:98]: we are working on it, but so far it is a manual process.

File: qdrant-landing/content/documentation/faq/qdrant-fundamentals.md (hunk 4 above)
Comment (generall) [reply qdrant-landing/content/documentation/faq/qdrant-fundamentals.md:91]: I am not sure if this is correct. Data parallelism might require multiple copies of the model per each process. But it have nothing to do with splitting the list of texts

File: qdrant-landing/content/documentation/faq/database-optimization.md (hunk 10 above)
Comment (kartik-gupta-ij) [reply qdrant-landing/content/documentation/faq/database-optimization.md:61]: basically to about what to index and what to not.

**Review by davidmyriel — APPROVED** [review]

**Pull request merged**